bpf/flows.c

```diff
@@ -79,6 +80,17 @@ static inline void update_dns(additional_metrics *extra_metrics, pkt_info *pkt,
 }
 
 static inline int flow_monitor(struct __sk_buff *skb, u8 direction) {
+    u32 filter_sampling = 0;
+
+    if (!is_filter_enabled()) {
+        if (sampling > 1 && (bpf_get_prandom_u32() % sampling) != 0) {
+            do_sampling = 0;
+            return TC_ACT_OK;
+        }
+        filter_sampling = sampling;
+        do_sampling = 1;
+    }
```

Comment:
What should `do_sampling` be set to if filter is enabled?  It is currently left unchanged.

Reply:
do_sampling is just bool either 0 or 1 sampling rate is what prandom uses

Reply:
Prior to this code change, `do_sampling` was always set to 0 or 1.  Now it can be left unchanged if `is_filter_enabled()` is false.  There are a few places in other parts of the code that does something depending on the state of `do_sampling`.  See: [https://github.com/search?q=repo%3Anetobserv/netobserv-ebpf-agent%20do_sampling&type=code](https://github.com/search?q=repo%3Anetobserv/netobserv-ebpf-agent%20do_sampling&type=code).

Reply:
@stleerh I think all code paths are covered to set a value on do_sampling:
- if filters aren't enabled (line 85), do_sampling is set to either 0 or 1
- if filters are enabled (line 118), do_sampling is also set to either 0 or 1

The execution between line 85 and 118 are just IP/TCP headers extraction, they won't involve checking do_sampling, so that should be fine

Reply:
Okay, I didn't see that line 85 is not the same as line 118.

/lgtm